### PR TITLE
cmd/atlas/internal/migrate: revisions work on same connection as the migration execution

### PR DESCRIPTION
**Pull request by masseelch**
This enables the revision access work within the same transaction as the execution. This way we ensure atomicity on the database connection and correctness for the revision information to match the actual state of the database.

Logging information about migration attempts, rollbacks, errors and stuff like this will be separated from the actual revision information and be done somewhere else (e.g. nebula) and this will follow in another PR.

Since this PR was required for the `--tx-mode` flag to work properly the missing test for SQLite has been added as well.

### CODE REVIEW

**Review by ronenlu — APPROVED** [review]

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 1]
```diff
@@ -354,40 +346,41 @@ func CmdMigrateApplyRun(cmd *cobra.Command, args []string) error {
 		return err
 	}
 	var (
-		mux = tx{ctx: cmd.Context(), c: c, rrw: rrw, files: pending}
+		tx  = tx{c: c, rrw: rrw}
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/migrate.go:349]: Try to avoid this shadowing.

File: cmd/atlas/internal/migrate/migrate.go [hunk 2]
```diff
@@ -24,19 +25,21 @@ const DefaultRevisionSchema = "atlas_schema_revisions"
...
 	Option func(*EntRevisions) error
 )
 
-// NewEntRevisions creates a new EntRevisions with the given sqlclient.Client. It is important to call
-// EntRevisions.Init to initialize the underlying Ent client.
-func NewEntRevisions(ac *sqlclient.Client, opts ...Option) (*EntRevisions, error) {
+// NewEntRevisions creates a new EntRevisions with the given sqlclient.Client.
+func NewEntRevisions(ctx context.Context, ac *sqlclient.Client, opts ...Option) (*EntRevisions, error) {
+	// MySQL connections need to set the parseTime value.
+	if ac.Name == dialect.MySQL && strings.ToLower(ac.URL.Query().Get("parseTime")) != "true" {
```
Comment (a8m) [inline cmd/atlas/internal/migrate/migrate.go:40]: Let’s handle it on Open. It’s weird to require this from users.

File: cmd/atlas/internal/migrate/migrate.go [hunk 3]
```diff
@@ -119,78 +110,37 @@ func (r *EntRevisions) ReadRevisions(ctx context.Context) ([]*migrate.Revision,
...
-	// the database until the transaction wrapping the migration execution has been committed.
-	return r.ac.Name == dialect.SQLite
-}
-
-func (r *EntRevisions) openSchema(ctx context.Context) (*sqlclient.Client, error) {
-	u := r.ac.URL.URL
-	if r.ac.Name == dialect.MySQL && !u.Query().Has("parseTime") {
-		v := *u
-		q := v.Query()
-		q.Set("parseTime", "true")
```
Comment (a8m) [inline cmd/atlas/internal/migrate/migrate.go:189]: See my comment above.

File: internal/integration/mysql_test.go [hunk 4]
```diff
@@ -1278,7 +1278,7 @@ func (t *myTest) dsn(dbname string) string {
 	if t.mariadb() {
 		d = "mariadb"
 	}
-	return fmt.Sprintf("%s://root%s@localhost:%d/%s", d, pass, t.port, dbname)
```
Comment (a8m) [inline internal/integration/mysql_test.go:1281]: Remove and let Atlas handle it for the users.

**Review by a8m — APPROVED** [review]
💯

File: internal/integration/mysql_test.go (hunk 4 above)
Comment (masseelch) [reply internal/integration/mysql_test.go:1281]: This is in the test without a client open in between. Here it is valid to add the parseTime by hand.

**Pull request merged**